### PR TITLE
Move contract types to their own package

### PR DESCRIPTION
**Pull request by nalinbhardwaj**
We need some of them for client-side app as well, for building userops and submitting transactions

### CODE REVIEW

**Comment by nalinbhardwaj** [comment]
Also tried adding a test to make sure the generated file is never out of date with the latest run files, surprisingly it seems `wagmi generate` does not use stable sort for the file types (so there's always a diff)

**Comment by dcposch** [comment]
I ran into the `wagmi generate` shuffle issue, too

**Comment by dcposch** [comment]
### Filed an issue for `wagmi generate` stable output

https://github.com/wagmi-dev/wagmi/issues/2528

Quick response from the devs 🚀

File: packages/contract-types/src/index.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export * as Contracts from "./generated";
```
Comment (dcposch) [inline packages/contract-types/src/index.ts:1]: 🚀

File: packages/contract-types/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    "module": "commonjs",
+    "esModuleInterop": true,
+    "target": "es2019",
+    "moduleResolution": "node",
+    "sourceMap": true,
+    "outDir": "dist",
+    "resolveJsonModule": true
+  },
+  "lib": ["es2019"]
+}
```
Comment (dcposch) [inline packages/contract-types/tsconfig.json:12]: you can use the node 20 preset:

```
{
  "extends": "@tsconfig/node20/tsconfig.json",
  "compilerOptions": {
    "strict": true,
    "resolveJsonModule": true,
    "sourceMap": true,
    "outDir": "dist"
  }
}
```

either way, most important is to add `strict: true`

File: packages/daimo-api/package.json [hunk 3]
```diff
@@ -18,7 +17,7 @@
     "@tsconfig/node20": "^1.0.1",
     "@types/tape": "^5.6.0",
     "@wagmi/chains": "^1.1.0",
-    "@wagmi/cli": "^1.1.0",
+    "daimo-contract-types": "file:../contract-types",
```
Comment (dcposch) [inline packages/daimo-api/package.json:20]: 🚀 

good call

**Review by dcposch — APPROVED** [review]
LGreatTM

File: packages/contract-types/tsconfig.json (hunk 2 above)
Comment (dcposch) [reply packages/contract-types/tsconfig.json:12]: ^ i'll add it in a follup. merging now

Comment (nalinbhardwaj) [reply packages/contract-types/tsconfig.json:12]: fixed in https://github.com/daimo-eth/daimo/pull/24

**Pull request merged**